### DISCUSSION TITLE
Terraform Source: Private GitHub Repo or Git URL

### DISCUSSION THREAD

**Discussion by ekristen** [renovatebot/renovate#8718] [Request Help] [answered]
Is it possible to have Renovate find and detect updates for private github repositories or github repositories when using the full git url syntax?

```hcl
module "label" {
  source = "git::https://github.com/cloudposse/terraform-null-label.git?ref=tags/0.19.2"
}
```

OR 

```hcl
module "label" {
  source = "git::ssh://git@github.com/ekristen/terraform-aws-bastion.git?ref=0.1.0"
}
```

Documentation says github is supported, but I suspect it's just `github.com/ekristen/terraform-aws-bastion?ref=0.1.0` and has to be public.

**Comment by rarkins** [comment, marked as answer]
The regex matching is quite flexible, so I think it should work: https://github.com/renovatebot/renovate/blob/68f89c48165d67a40933e448e393611ef5a3f81a/lib/manager/terraform/modules.ts#L11

And it doesn't need to be public - it will try to query the repo regardless, and if the token you have provided has permissions to access the private repo then it will work

**Reply by ekristen** [reply]
As always thank you for your lightning fast reply. Quick follow up, to which token are you referring?

**Reply by rarkins** [reply]
It depends on how you're running it. If you're in the hosted app, then the token will have permissions to access all private repos within the same org (but only if the repo it's running on is private too).
